felix/rules/static.go

```diff
@@ -1002,11 +1002,58 @@ func (r *DefaultRuleRenderer) StaticBPFModeRawChains(ipVersion uint8, tcBypassMa
 	rawPreroutingChain := &Chain{
 		Name: ChainRawPrerouting,
 		Rules: []Rule{
+			Rule{
+				Match:  Match().NotDestAddrType(AddrTypeLocal),
+				Action: GotoAction{Target: ChainRawUntrackedFlows},
+			},
 			Rule{
 				// Return, i.e. no-op, if bypass mark is not set.
-				Match:  Match().NotMarkMatchesWithMask(tcBypassMark, 0xffffffff),
-				Action: ReturnAction{},
+				Match:   Match().MarkMatchesWithMask(tcBypassMark, 0xffffffff),
+				Action:  GotoAction{Target: ChainRawBPFUntrackedPolicy},
+				Comment: []string{"Jump to target for packets with Bypass mark"},
 			},
+		},
+	}
+
+	bpfUntrackedFlowChain := &Chain{
+		Name: ChainRawUntrackedFlows,
+		Rules: []Rule{
+			Rule{
+				Match:   Match().MarkMatchesWithMask(tcdefs.MarkSeenSkipFIB, tcdefs.MarkSeenSkipFIB),
+				Action:  ReturnAction{},
+				Comment: []string{"MarkSeenSkipFIB Mark"},
+			},
+			Rule{
+				Match:   Match().MarkMatchesWithMask(tcdefs.MarkSeenFallThrough, tcdefs.MarkSeenFallThroughMask),
+				Action:  ReturnAction{},
+				Comment: []string{"MarkSeenFallThrough Mark"},
+			},
+			Rule{
+				Match:   Match().MarkMatchesWithMask(tcdefs.MarkSeenMASQ, tcdefs.MarkSeenMASQMask),
+				Action:  ReturnAction{},
+				Comment: []string{"MarkSeenMASQ Mark"},
+			},
+			Rule{
+				Match:   Match().MarkMatchesWithMask(tcdefs.MarkSeenNATOutgoing, tcdefs.MarkSeenNATOutgoingMask),
+				Action:  ReturnAction{},
+				Comment: []string{"MarkSeenNATOutgoing Mark"},
+			},
+			Rule{
+				Match:   Match().MarkMatchesWithMask(tcdefs.MarkSeenBypass, tcdefs.MarkSeenBypassMask),
+				Action:  ReturnAction{},
+				Comment: []string{"MarkSeenBypass Mark"},
+			},
+			Rule{
+				Match:   Match().MarkMatchesWithMask(tcdefs.MarkSeen, tcdefs.MarkSeenMask),
```

Comment:
do we need to test the seen? We drop all traffic that does not have seen anyway.

Reply:
Ok. I removed the match since it looks more aligned with the ideal behaviour.